crates/compiler/build/src/program.rs

```diff
@@ -260,7 +259,19 @@ fn gen_from_mono_module_llvm<'a>(
     // Uncomment this to see the module's optimized LLVM instruction output:
     // env.module.print_to_stderr();
 
-    mpm.run_on(module);
+    let inkwell_opt_level = crate::target::convert_opt_level(opt_level);
+    let inkwell_llvm_passes = crate::llvm_passes::get_llvm_passes_str(opt_level);
+    let inkwell_target_machine =
+        crate::target::target_machine(target, inkwell_opt_level, inkwell::targets::RelocMode::PIC)
+            .expect("should be a valid target machine");
+
+    module
+        .run_passes(
+            inkwell_llvm_passes,
+            &inkwell_target_machine,
+            inkwell::passes::PassBuilderOptions::create(),
+        )
+        .expect("valid llvm optimization passes");
```

Comment:
We have replaced all the locations where we use llvm optimisations with the same helpers in roc_build to get the inkwell/llvm target_machine and llvm_passes.